tests/lib/utils/ast-utils/extract-expression-references.ts

```diff
@@ -114,72 +114,96 @@ describe("extractExpressionReferences", () => {
-                            const scope = context.sourceCode.getScope(program)
-                            const tracker = new ReferenceTracker(scope)
-
-                            for (const {
-                                node,
-                            } of tracker.iterateGlobalReferences({
-                                RegExp: { [CALL]: true, [CONSTRUCT]: true },
-                            })) {
-                                const newOrCall = node as
-                                    | ESTree.NewExpression
-                                    | ESTree.CallExpression
-                                regexps.push(newOrCall)
-                            }
-                        },
-                        Literal(node: ESTree.Literal) {
-                            if (isRegexpLiteral(node)) {
-                                testNodes.push(node)
-                            }
-                        },
-                        "NewExpression,CallExpression"(
-                            node: ESTree.NewExpression | ESTree.CallExpression,
-                        ) {
-                            if (regexps.includes(node)) {
-                                testNodes.push(node)
-                            }
-                        },
-                        "Program:exit"() {
-                            results = testNodes.map((node) =>
-                                [
-                                    ...extractExpressionReferences(
-                                        node,
-                                        context,
-                                    ),
-                                ].map((r) => toResult(r, context)),
-                            )
-                        },
-                    }
-                },
-            })
-            const r = linter.verify(
+            const linter = new Linter({ configType: "flat" })
+            const result = linter.verify(
                 testCase.code,
                 {
-                    globals: {
-                        Set: "readonly",
-                        Map: "readonly",
-                        BigInt: "readonly",
-                        window: "readonly",
-                        globalThis: "readonly",
+                    plugins: {
+                        // @ts-expect-error -- ignore type error for eslint v9
+                        test: {
+                            rules: {
+                                test: {
+                                    create(context: Rule.RuleContext) {
+                                        return {
+                                            Program(program: ESTree.Program) {
+                                                const scope =
+                                                    context.sourceCode.getScope(
+                                                        program,
+                                                    )
```

Comment:
We are hitting the character limit hard here. Maybe we could extract the `create` function out into a variable and put it above `const linter = new Linter({ configType: "flat" })`?